src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -252,6 +477,215 @@ private TemporaryPoint createNewPoint(final Coordinate coordinate,
         return tags;
     }
 
+    /**
+     * Creates a mapping of all inner rings enclosed by the outer rings.
+     *
+     * @param outerToInnerMap
+     *            The mapping to update any time we find an outer ring that contains an inner ring
+     * @param outerRings
+     *            The list of outer {@link LinearRing}s to use
+     * @param innerRings
+     *            The list of inner {@link LinearRing}s to use
+     */
+    private void findEnclosedRings(final MultiMap<Integer, Integer> outerToInnerMap,
+            final List<LinearRing> outerRings, final List<LinearRing> innerRings)
+    {
+        for (int innerIndex = 0; innerIndex < innerRings.size(); innerIndex++)
+        {
+            final LinearRing inner = innerRings.get(innerIndex);
+            boolean foundEnclosingOuter = false;
+            for (int outerIndex = 0; outerIndex < outerRings.size(); outerIndex++)
+            {
+                final LinearRing outer = outerRings.get(outerIndex);
+                final com.vividsolutions.jts.geom.Polygon outerPolygon = new com.vividsolutions.jts.geom.Polygon(
+                        outer, null, JtsUtility.GEOMETRY_FACTORY);
+                if (outerPolygon.contains(inner))
+                {
+                    foundEnclosingOuter = true;
+                    outerToInnerMap.add(outerIndex, innerIndex);
+                }
+            }
+
+            if (!foundEnclosingOuter)
+            {
+                // Isolated inner ring, invalid multipolygon
+                logger.error("Found isolated inner member for Multipolygon Relation geometry: {}",
+                        inner);
+            }
+        }
+    }
+
+    /**
```

Comment:
This is more a personal stylistic thing for me, but when you add java docs for a method, I think it is good to include a description of the method above. I mean in this particular instance the method name is fairly self explanatory, but I think it should include a brief description for clarity and style.

Reply:
Yeah, I see what you mean. I just thought it might be a bit repetitive to add javadoc description as well as the return field. Will update it though, to be consistent with the other method javadocs.